lib/timeago.rb

```diff
@@ -0,0 +1,31 @@
+class TimeAgo
+
+  def self.run(time)
+    TimeAgo.new.run(time)
+  end
+
+  def run(time)
+    time_ago = largest_denomination(time)
+    "#{time_ago.last} #{time_ago.first}#{time_ago.last > 1 ? 's' : ''}"
+  end
+
+  private
+  DENOMINATIONS = {day: 86_400,
+                   hour: 3600,
+                   minute: 60,
+                   second: 1}
+
+  def largest_denomination(time)
+    time_by_denomination(time).select{|denomination, amount| amount > 0}.first
```

Comment:
Lint/UnusedBlockArgument: Unused block argument - `denomination`. If it's necessary, use `_` or `_denomination` as an argument name to indicate that it won't be used. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)<br>Performance/Detect: Use `find` instead of `select.first`. (https://github.com/JuanitoFatas/fast-ruby#enumerabledetect-vs-enumerableselectfirst-code)